File: bench/data-structures-bst/1.go
package main

import (
	"math/rand"
	"os"
	"strconv"
	"fmt"
	"time"

	"github.com/google/btree"
)

func main() {
	rand.Seed(time.Now().UnixNano())

	var n = 1000000
	var args = os.Args[1:]
	if len(args) >= 1 {
		n, _ = strconv.Atoi(args[0])
	}

	var tr = btree.New(2)

	var start = time.Now()

	for i := 0; i < n; i++ {
		tr.ReplaceOrInsert(btree.Int(rand.Int()))
	}

	fmt.Println(time.Now().Sub(start).Nanoseconds())
}

File: bench/data-structures-bst/1.cpp
#include <iostream>
#include <ctime>
#include <cstdlib>
#include "cpp-btree/btree_set.h"
#include <chrono>

using namespace std;

int main(int argc, char const *argv[])
{
    srand(time(NULL));

	btree::btree_set<int> btree;

	int n = 1000000;
	if (argc == 2)
	{
		n = atoi(argv[1]);
	}

	auto start = chrono::steady_clock::now();

	for (int i = 0; i < n; i++)
	{
		btree.insert(rand());
	}

	cout << chrono::duration_cast<chrono::nanoseconds>(chrono::steady_clock::now() - start).count() << endl;

    return 0;
}
